descriptions.md

```diff
@@ -306,12 +306,12 @@ Enables messages to be published and subscribed to. Also enables historic messag
 | publish(name: String?, data: Data?) => io ||| RTL6i | Publishes a single message to the channel with the given event name and payload. A callback may optionally be passed in to this call to be notified of success or failure of the operation. When publish is called with this client library, it won't attempt to implicitly attach to the channel, so long as [transient publishing](https://ably.com/docs/realtime/channels#transient-publish) is available in the library. Otherwise, the client will implicitly attach. |
 || `name` ||| The event name. |
 || `data` ||| The message payload. |
-| subscribe((Message) ->) => io ||| RTL7a | Registers a listener for messages on this channel. The caller supplies a listener function, which is called each time one or more messages arrives on the channel. A callback may optionally be passed in to this call to be notified of success or failure of the channel [`attach()`]{@link RealtimeChannel#attach} operation. |
+| subscribe((Message) ->) => io ||| RTL7a | Registers a listener for messages on this channel. The caller supplies a listener function, which is called each time one or more messages arrives on the channel. A callback may optionally be passed in to this call to be notified of success or failure of the channel [`attach()`]{@link RealtimeChannel#attach} operation. It will not be called if [`attachOnSubscribe`]{@link ChannelOptions#attachOnSubscribe} channel option was set to `false`. |
```

Comment:
```suggestion
| subscribe((Message) ->) => io ||| RTL7a | Registers a listener for messages on this channel. The caller supplies a listener function, which is called each time one or more messages arrives on the channel. A callback may optionally be passed in to this call to be notified of success or failure of the channel [`attach()`]{@link RealtimeChannel#attach} operation. It will not be called if the [`attachOnSubscribe`]{@link ChannelOptions#attachOnSubscribe} channel option is set to `false`. |
```

(Ditto the other similar docstrings added)

Reply:
I am not sure if this makes sense : (. 
If optional callback is passed to denote success of the operation ( as per specified options ), callback should get called. In case we are mentioning explicitly not to attach, callback should be called.
We are doing the same in case of ably-java

Reply:
In the end, optional callback is provided to denote success of the given operation. It should get called if specified operation is performed.

Reply:
If it's never getting called, then it's a questionable param

Reply:
But how calling `onAttach` callback without actual attach operation makes more sense? Maybe call it with not critical error like "Attach operation wasn't performed due to options value" or smth?

Reply:
>(Ditto the other similar docstrings added)

done

Reply:
@sacOO7 the behaviour that Marat has described here (not calling the callback) is specified by [RTL7h](https://sdk.ably.com/builds/ably/specification/main/features/#RTL7h) and [RTP6e](https://sdk.ably.com/builds/ably/specification/main/features/#RTP6e); if you disagree with this behaviour then please propose a change to the spec.